File: packages/react-native-worklets/android/src/worklets/main/com/swmansion/worklets/AnimationFrameCallback.java
package com.swmansion.worklets;

import com.facebook.jni.HybridData;
import com.facebook.proguard.annotations.DoNotStrip;

/**
 * @noinspection JavaJniMissingFunction
 */
@DoNotStrip
public class AnimationFrameCallback {

  @DoNotStrip private final HybridData mHybridData;

  @DoNotStrip
  private AnimationFrameCallback(HybridData hybridData) {
    mHybridData = hybridData;
  }

  public native void onAnimationFrame(double timestampMs);
}

File: packages/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/nativeProxy/AnimationFrameCallback.kt
package com.swmansion.reanimated.nativeProxy

import com.facebook.jni.HybridData
import com.facebook.proguard.annotations.DoNotStrip
import com.swmansion.reanimated.NodesManager

@DoNotStrip
class AnimationFrameCallback : NodesManager.OnAnimationFrame {

  @field:DoNotStrip private val mHybridData: HybridData

  @DoNotStrip
  private constructor(hybridData: HybridData) {
    mHybridData = hybridData
  }

  @DoNotStrip
  external override fun onAnimationFrame(timestampMs: Double)
}
